docs/src/transactions/transaction-execution.md

```diff
@@ -0,0 +1,52 @@
+# Transaction Execution
+Transactions are being executed by the Miden Transaction Executor. Transaction execution results in a `ExecutedTransaction` object and consists of the following steps:
+
+1. Fetch the data required to execute a transaction from the `DataStore`.
+2. Compile the transaction into an executable [MASM](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) program using the `TransactionCompiler`.
+3. Execute the transaction program and create an `ExecutedTransaction`.
```

Comment:
I would create simple diagram to illustrate this.

Also, I would make this about the whole process starting from compilation/execution to proving. One of the main reasons for this design is that it allows us to have "stateless provers" - i.e., the executed transaction contains all data needed to re-execute and prove a transaction (no database access is needed). This is very powerful and allows us to distribute proof generation much more easily.